aws/resource_aws_elasticsearch_domain.go

```diff
@@ -633,6 +679,19 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 		return err
 	}
 
+	// Use AdvancedSecurityOptions from resource if possible
+	// because DescribeElasticsearchDomainConfig does not return MasterUserOptions
+	if ds.AdvancedSecurityOptions != nil {
+		if _, ok := d.GetOk("advanced_security_options"); ok {
+			d.Set("advanced_security_options.0.enabled", ds.AdvancedSecurityOptions.Enabled)
+			d.Set("advanced_security_options.0.internal_user_database_enabled", ds.AdvancedSecurityOptions.InternalUserDatabaseEnabled)
+		} else {
+			if err := d.Set("advanced_security_options", flattenAdvancedSecurityOptions(ds.AdvancedSecurityOptions)); err != nil {
+				return fmt.Errorf("error setting advanced_security_options: %s", err)
```

Comment:
Nit: my fault here in the example I provided 😅 do you mind switching this to the new error wrapping in Go 1.13 ("%w"), rather than the string value of the original error?